File: app/src/main/java/sgtmelon/scriptum/app/room/converter/StringConverter.java
package sgtmelon.scriptum.app.room.converter;

import android.text.TextUtils;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.room.TypeConverter;
import sgtmelon.scriptum.office.annot.DbAnn;

/**
 * Преобразование String - List<Long>
 * В строке разделителем является DIVIDER {@link DbAnn}
 */
public final class StringConverter {

    @TypeConverter
    public List<Long> fromString(@NonNull String value) {
        if (!value.equals(DbAnn.Value.NONE) && !value.equals("")) {
            final String[] stringArray = value.split(DbAnn.Value.DIVIDER);
            final List<Long> listLong = new ArrayList<>();

            for (String str : stringArray) {
                listLong.add(Long.parseLong(str));
            }

            return listLong;
        } else {
            return new ArrayList<>();
        }
    }

    @TypeConverter
    public String toString(List<Long> value) {
        if (value != null && value.size() != 0) {
            return TextUtils.join(DbAnn.Value.DIVIDER, value);
        } else {
            return DbAnn.Value.NONE;
        }
    }

}

File: app/src/main/java/sgtmelon/scriptum/app/room/converter/StringConverter.kt
package sgtmelon.scriptum.app.room.converter

import androidx.room.TypeConverter
import sgtmelon.scriptum.app.model.key.DbField.Value
import sgtmelon.scriptum.app.model.key.DbField
import java.util.*

/**
 * Преобразование String - List<Long>
 * В строке разделителем является DIVIDER [DbField]
 */
class StringConverter {

    @TypeConverter
    fun toList(string: String): MutableList<Long> = ArrayList<Long>().apply {
        if (string != Value.NONE && string.isNotEmpty()) {
            string.split(Value.DIVIDER.toRegex())
                    .dropLastWhile { it.isEmpty() }
                    .forEach { add(it.toLong()) }
        }
    }

    @TypeConverter
    fun toString(list: List<Long>?) =
            if (list != null && list.isNotEmpty()) list.joinToString(Value.DIVIDER) else Value.NONE

}